File: 1849-maximum-absolute-sum-of-any-subarray/1849-maximum-absolute-sum-of-any-subarray.cpp
class Solution {
public:
    int maxAbsoluteSum(vector<int>& nums) {
        int currMaxSum = nums[0];
        int maxSum = nums[0];

        int currMinSum = nums[0];
        int minSum = nums[0];
    
        for(int i = 1; i < nums.size(); i++) {
            currMaxSum = max(currMaxSum + nums[i], nums[i]);
            maxSum = max(maxSum, currMaxSum);

            currMinSum = min(currMinSum + nums[i], nums[i]);
            minSum = min(minSum, currMinSum);
        }
        return max(abs(minSum), abs(maxSum));
    }
};

File: 1849-maximum-absolute-sum-of-any-subarray/1849-maximum-absolute-sum-of-any-subarray.java
class Solution {
    public int maxAbsoluteSum(int[] nums) {
        int currMaxSum = nums[0];
        int maxSum = nums[0];

        int currMinSum = nums[0];
        int minSum = nums[0];

        for(int i = 1; i < nums.length; i++) {
            currMaxSum = Math.max(currMaxSum + nums[i], nums[i]);
            maxSum = Math.max(currMaxSum, maxSum); 

            currMinSum = Math.min(currMinSum + nums[i], nums[i]);
            minSum = Math.min(currMinSum, minSum);
        }
        return Math.max(Math.abs(maxSum), Math.abs(minSum));
    }
}